articles/creating-a-location-tracking-app-using-firebase-and-google-maps-in-android/index.md

```diff
@@ -0,0 +1,386 @@
+
+## Introduction
+
+In this tutorial, you will learn the steps in creating an application that uses Google Maps to give the precise location of another user's device. You shall primarily use Google Maps API to show the exact location. This will be achieved by writing the user's location on a Realtime Database using Firebase and read the location from another device.
+
+Location services have had many applications, especially social apps such as fitness apps, Pokemon Go. And hey, you can also find the location of your stolen device.
+
+By the end of this tutorial, you shall learn how to create two simple apps to aid with location tracking, how to work with Google Maps API and Firebase's Realtime Database.
+
+Comments will be added to the code snippets for better understanding.
+
+### Pre-requisites
+- A good understanding of Kotlin in Android development
+
+#### Reccomendation
+For this tutorial, we shall use real Android device(s) to test the application(s). There might arise a reason why you might test them using an emulator. This is achieved by mocking location data in the emulator. You can read more [here](https://medium.com/@msaudi/android-test-location-services-and-gps-with-fake-gps-data-mock-locations-in-emulators-or-real-df211de4d891)
+
+### Getting started
+##### Step 1: Create a new project
+Open Android Studio and create a new project. On the Select, a Project template, select the Google Maps Activity template
+[project template](/engineering-education/creating-a-location-tracking-app-using-firebase-and-google-maps-in-android/project-template.png)
+
+Wait for Android studio to build your project.
+
+At this point when you run the app, you have a blank screen. This is because you have not set up the API key for the map.
+
+## A few things to note
+Once you open the `AndroidManifest.xml`, you shall be able to note these auto-filled details:
+- `ACCESS_FINE_LOCATION` permission. This accesses the user's precise location. it is often used when you want the most accurate location. Another type of location accuracy is the `ACCESS_COARSE_LOCATION`, which is less accurate. We shall be working with the `ACCESS_FINE_LOCATION` for accurate readings.
+- The `com.google.android.geo.API_KEY` meta-data. This is used to specify the API key
+
+The template also adds Google Play Services dependency to build.gradle which exposes the API to the application
```

Comment:
`exposes` seems to be incorrect in this statement

Reply:
The Google Play Services dependency actually exposes the API to the application. I do not have another way to explain that. If you could help please do. Thank you